File: java/3355.java
class Solution {
    public boolean isZeroArray(int[] n, int[][] q) {
        int l = n.length;
        int r[] = new int[l + 1];
        for (int i[] : q) {
            r[i[0]]++;
            r[i[1] + 1]--;
        }
        int p = 0;
        for (int i = 0; i < l; i++) {
            p += r[i];
            if (n[i] > p)
                return false;
        }
        return true;
    }
}

File: python3/3355.py
class Solution:
    def isZeroArray(self, n: List[int], q: List[List[int]]) -> bool:
        l = len(n)
        r = [0] * (l + 1)
        p = 0
        for a, b in q:
            r[a] += 1
            r[b + 1] -= 1
        for i in range(l):
            p += r[i]
            if n[i] > p:
                return False
        return True
